src/SmartTransactionsController.ts

```diff
@@ -47,165 +48,237 @@ import {
 
 const SECOND = 1000;
 export const DEFAULT_INTERVAL = SECOND * 5;
+const DEFAULT_CLIENT_ID = 'default';
 const ETH_QUERY_ERROR_MSG =
   '`ethQuery` is not defined on SmartTransactionsController';
 
-export type SmartTransactionsControllerConfig = BaseConfig & {
-  interval: number;
-  clientId: string;
-  chainId: Hex;
-  supportedChainIds: Hex[];
+/**
+ * The name of the {@link SmartTransactionsController}
+ */
+const controllerName = 'SmartTransactionsController';
+
+const controllerMetadata = {
+  smartTransactionsState: {
+    persist: false,
+    anonymous: true,
+  },
 };
```

Comment:
Hmm, putting the whole controller state under the `smartTransactionsState` forces us to apply the same metadata to all of it. I know that this was already here before this PR, but thoughts on removing this key and putting the entire state as root object?

Reply:
This will assume there will be a need for a migration script we can maybe create an issue for this and address this change in a separate ticket